src/components/web-monitoring-ui.jsx

```diff
@@ -48,52 +61,113 @@ export default class WebMonitoringUi extends React.Component {
+    this.loadPages('pages');
   }
 
-  loadPages () {
+  /**
+   * Load pages depending on whether we want all pages or assigned pages.
+   * @private
+   * @param {string} pageFilter Must be either 'assignedPages' or 'pages'
+   *
+   * Sends a requests out to either db-api or localApi for pages and sets
+   * corresponding `assignedPages` or `pages` property of state and `pageFilter`.
+   * These are passed as props to various child components.
+   */
+  loadPages (pageFilter) {
     api.isLoggedIn()
       .then(loggedIn => {
-        this.setState({user: api.userData});
+        if (!loggedIn) {
+          return Promise.reject(new Error('You must be logged in to view pages'));
+        }
+
         const query = {include_latest: true};
-        if (loggedIn) {
-          return localApi.getPagesForUser(api.userData.email, null, query)
-            .catch(() => {
-              // TODO: Handle 'user not found' in a better way
-              // than just showing default list
-              return api.getPages(query);
-            });
+        if (pageFilter === 'assignedPages') {
+          return localApi.getPagesForUser(api.userData.email, null, query);
         }
         else {
           return api.getPages(query);
         }
       })
-      .then((pages) => {
-        this.setState({pages});
+      .then(pages => {
+        this.setState({
+          [pageFilter]: pages,
+          pageFilter
+        });
+      });
+  }
+
+  loadUser () {
+    api.isLoggedIn()
+      .then(loggedIn => {
+        this.setState({user: api.userData, isLoading: false});
       });
   }
 
   componentWillMount () {
-    this.loadPages();
+    this.loadUser();
   }
 
   render () {
-    const withData = bindComponent({pages: this.state.pages, user: this.state.user});
+    if (this.state.isLoading) {
+      return <Loading />;
+    }
+
+    if (!this.state.user) {
+      return this.renderLoginDialog();
+    }
+
+    const withData = (ComponentType, pageType) => {
+      return (routeProps) => {
+        const pages = this.state[pageType];
+        if (!pages) {
+          this.loadPages(pageType);
+        }
+        return <ComponentType
+          {...routeProps}
+          pages={pages}
+          user={this.state.user}
+        />;
+      };
+    };
     const modal = this.state.showLogin ? this.renderLoginDialog() : null;
 
     return (
       <div>
         <Router>
           <div id="application">
-            <NavBar title="EDGI" user={this.state.user} showLogin={this.showLogin} logOut={this.logOut} />
-            <Route exact path="/" render={withData(PageList)} />
-            <Route path="/page/:pageId/:change?" render={withData(PageDetails)} />
+            <NavBar
+              title="EDGI"
+              user={this.state.user}
+              showLogin={this.showLogin}
+              logOut={this.logOut}
+              pageFilter={this.state.pageFilter}
+              setPageFilter={this.setPageFilter}
+            />
+            <div>
```

Comment:
Do we need this `div`?

Reply:
Nope, sorry, this was an artifact from when I was trying to figure out how to work with react-router. Removed